src/neo4j_genai/types.py

```diff
@@ -27,17 +27,57 @@
 from neo4j_genai.retrievers.utils import validate_search_query_input
 
 
-class VectorSearchRecord(BaseModel):
+class RetrieverResultItem(BaseModel):
     """
-    Represents a record returned from a vector search.
+    A single record returned from a retriever.
 
     Attributes:
-        node (Any): The node data retrieved by the search.
-        score (float): The similarity score of the retrieved embedding.
+        content (str): The context as will be provided to the LLM
+        metadata (Optional[dict]): Any metadata that can be included together
+            with the text, related to that record (e.g. another node property)
     """
 
-    node: Any
-    score: float
+    content: str
+    metadata: Optional[dict] = None
+
+
+class RetrieverRawResult(BaseModel):
+    """
+    Represents the raw result returned from the retriever get_search_result
+    method. It needs to be formatted further before being returned as a RetrieverResult.
+
+    Attributes:
+        records (list[neo4j.Record]): A list of records from neo4j.
+        metadata: Record-related metadata, such as score.
+    """
+
+    records: list[neo4j.Record]
+    metadata: Optional[dict] = None
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
Is model_config relevant to retrieved result? If not, it could be better to some other class.

Reply:
`model_config` is a Pydantic parameter. We use it to tell pydantic not to perform type checking (because `neo4j.*` do not implement the required methods for that)